tests/unit/transform-value-to-date-or-range.spec.ts

```diff
@@ -92,6 +92,7 @@ describe('transformRelativeDateRangeToDateRange', () => {
       date: '{{toto}}',
       quantity: -2,
```

Comment:
:warning: Should we remove in tests all traces of cases where the quantity is negative?
As the operator is now used to determine the order of base and target, these case will produce intervals where the end precedes the beginning?

Reply:
> As the operator is now used to determine the order of base and target

Full reasoning is "the operator is now used to determine the quantity sign which is used to determine the order of base and target.
I'm taking shortcut in the code but add a intermediate line to make it clear.

I don't see why this should impact tests in the end :shrug:

Reply:
I didn't see any modifications in `https://github.com/ToucanToco/weaverbird/pull/1177/files#diff-f9f2b6ed3933ed5ca25d0627cfbad7ecc4e88799f09904ee7eba4afd3471daffR77-R80` (https://github.com/ToucanToco/weaverbird/pull/1177/files#diff-f9f2b6ed3933ed5ca25d0627cfbad7ecc4e88799f09904ee7eba4afd3471daffR77-R80) in your PR, so I assume that the sign is still used to compute the target date.

By "the operator is now used to determine the quantity sign ", I was expecting a modification in this function that computes the target date, but reading again I realize this is done way before, before the value is emitted from the component (`Math.sign(operator.sign`).

So that means we have a redundant info between the operator and the sign of the quantity. Is that OK and easy to understand for later readers?

Reply:
> Is that OK and easy to understand for later readers?

Depend of the reader :grimacing: 
I get what you mean. Because the info is redundant, today is does not matter if we use the quantity sign or the operator sign.

Getting rid of this redundancy would mean that quantity is a positive integer that always need to have the sign applied to it. We can go this way :shrug: 

As TypeScript don't have any positive integer type, it will still be a precondition only "enforced" here and there.
Properly enforcing it require a type constructor & private members aka a good old Class.
We almost never use Classes but should consider them more often IMHO.

Reply:
> Getting rid of this redundancy would mean that quantity is a positive integer that always need to have the sign applied to it. We can go this way shrug

I would feel this matches the most the UI provided (we never ask our users to enter a negative quantity, but instead they do select an operator).
Still, I get that this clarity is somewhat subjective and I won't enforce you to do it before getting this PR approved.

Thanks for the clear answer and understanding :)

Reply:
> this matches the most the UI

the saved object also read better : 
`{ quantity: -2, duration: 'month', operator: 'until', date: '{{today}}' }` feel a little wrong compared to
`{ quantity: 2, duration: 'month', operator: 'until', date: '{{today}}' }`

So currently trying to refactor it more, but deep down I know that this will probably never matter :no_mouth: